pyuvdata/uvcal/calfits.py

```diff
@@ -90,19 +114,52 @@ def write_calfits(
                     "because of a select operation). The calfits format "
                     "does not support unevenly spaced times."
                 )
-            if np.isclose(time_spacing[0], self.integration_time / (24.0 * 60.0 ** 2)):
-                time_spacing = self.integration_time / (24.0 * 60.0 ** 2)
+            if self.future_array_shapes:
+                if not np.isclose(
```

Comment:
Just a suggestion, but since we're porting things over into `utils`, this may be another snippet of code (everything within the `np.isclose` call) worth moving to either the utilities or adding as a method of `UVParameter`, since I've seen it come up in many, many other places.

Reply:
Good idea. I made a new fairly generic utils function called `_test_array_constant` to test if an array contains all the same value to some tolerance. It accepts either UVParameters or numpy arrays. I went through the code base and tried to find all the places where we were doing something like this and replace them with calls to that function, but it's possible I missed some.